sqlmesh/core/engine_adapter/base.py

```diff
@@ -1747,12 +1754,44 @@ def remove_managed_columns(
                 # Historical Records that Do Not Change
                 .with_(
                     "static",
-                    existing_rows_query.where(valid_to_col.is_(exp.Null()).not_()),
+                    existing_rows_query.where(valid_to_col.is_(exp.Null()).not_())
+                    if truncate
+                    else existing_rows_query.where(
+                        exp.and_(
+                            valid_from_col <= cleanup_ts,
```

Comment:
Is this needed? The other condition is that `valid_to_col < cleanup_ts` which means `valid_from_col` would also be `<= cleanupo_ts` right?